Blogpost for the switch to Libera (#232)
TeamNewPipe/website: _posts/2021-05-23-irc-move.md

---
layout: post
title: "NewPipe bides farewell to freenode and can now be found on Libera.Chat"
short: "NewPipe moves to Libera.Chat"
date: 2021-05-23 09:00:00 + 1:00
categories: [pinned, release]
author: Opusforlife & Poolitzer
image: newpipe
excerpt_separator: <!-- more -->
---

We wrote an in-depth explanation below, but if you do not need or care about this, for you in a few words: Newpipe has an active IRC channel where developers and users chat and can ask questions. This channel moves from the freenode to the Libera.Chat network. Until Matrix finalizes its bridge to the new network, the room on freenode will be active, but as soon as that happens, it will be shut down.
<!-- more -->

## What is IRC?

> In order to provide a bit of a background, what follows will be short explanation what all this is. If you are familiar with IRC, skip to the next section.

Reviewer: ```suggestion
In order to provide a bit of background, what follows is a short explanation of what this is all about. If you are familiar with IRC, skip to the next section.
```

> [freenode](https://freenode.net/) and [Libera.Chat](https://libera.chat) are IRC networks. IRC stands for Internet Relay Chat. Without diving too deep in technical explanations (if you care for this, read the [wikipedia article](https://en.wikipedia.org/wiki/Internet_Relay_Chat)), it is an instant communication standard for groups. You install an IRC app on your device (or use a webapp), those act as IRC clients, which you then need to point to an IRC server. Bigger IRC servers are called networks, because instead of just one server, they are a network of several servers (makes sense). The biggest and oldest IRC network is called freenode, and we as NewPipe used this network to host our channel. Up until this week.

Reviewer: ```suggestion
[freenode](https://freenode.net/) and [Libera.Chat](https://libera.chat) are IRC networks. IRC stands for Internet Relay Chat. Without diving too deep into technical explanations (if you care for this, read the [wikipedia article](https://en.wikipedia.org/wiki/Internet_Relay_Chat)), it is an instant communication standard for groups. You install an IRC client app on your device (or use a webapp), which you then need to point to an IRC server. Bigger IRC servers are called networks, because instead of just one server, they are a network of several servers (makes sense). The biggest and oldest IRC network is called freenode, and Team NewPipe used this network to host our channel. Until this week, this was perfectly fine and no one batted an eye regarding any of this.
```

## What has happened?

> This will be a very short explanation again. This time, we will condense what apparently has happened behind the scenes of freenode. If you want to paint your own picture, read [this post](https://gist.github.com/realrasengan/88549ec34ee32d01629354e4075d2d48) from one side of this drama, and [this resigning letter](https://gist.github.com/aaronmdjones/1a9a93ded5b7d162c3f58bdd66b8f491) from the other. There are many more resigning letters around, you can search for it.

Reviewer: ```suggestion
Again, this will be a very short explanation. This time, we will attempt to condense what has apparently happened behind the scenes at freenode. If you want to paint your own picture, read [this post](https://gist.github.com/realrasengan/88549ec34ee32d01629354e4075d2d48) from one side of this drama, and [this resigning letter](https://gist.github.com/aaronmdjones/1a9a93ded5b7d162c3f58bdd66b8f491) from the other. There are many more resigning letters around; you can search for them.
```

> In a nutshell, about a dozen volunteer developers behind freenode left the project. This happened because Andrew Lee bought freenode Limited in 2017, a company which was used as a legal representation of freenode, mostly for their freenode Live Conferences. While he promised to not obtain any influence over freenode itself, this apparently changed this week. Andrew and his team gained full controll over the network (as stated by their [twitter account](https://twitter.com/freenodestaff/status/1395046345145307140)), which lead to the volunteers resign. Whilste resigning, mosted of the volunteers stated they will now be at Libera.Chat.

Reviewer: ```suggestion
In a nutshell, about a dozen volunteer developers behind freenode left the project recently. This happened because Andrew Lee bought Freenode Limited in 2017, a company which was used as a legal representation of freenode, mostly for their freenode Live Conferences. While he promised not to obtain any influence over freenode itself, this apparently changed this week. Andrew and his team gained full control over the network (as stated by their [Twitter account](https://twitter.com/freenodestaff/status/1395046345145307140)), which led to the volunteers resigning. While resigning, most of the volunteers stated they will now be at Libera.Chat.
```

## Why is NewPipe moving?

> Honestly, we do not trust an American business man saying he won't do anything with our data. We do trust developers, we are developers ourself. When you do something to a project which leads to its core team resigning, and said core team starts a new project, we will join this new project. We like our IRC channels, they are the best way to communicate with each other while staying True to our privacy values. After discussing the events internally, we decide to switch the network, and we are in good company doing so. 

Reviewer: ```suggestion
Honestly, we do not trust an American businessman saying he won't do anything with our data. We do trust developers, as we are developers ourselves. When someone does something to a project which leads to its core team resigning, and said core team starts a new project, we prefer to join the new project. We like our IRC channels; they are the best way to communicate with each other while staying true to our privacy values. After discussing the events internally, we decided to switch networks, and we are in good company doing so.
```

Reviewer: What about mentioning examples of other major orgs that are switching, like Ubuntu's IRC channels?

Author: thought about it, not sure if it was fitting or not.

## What do I have to do?

You won't have to do much if you use an IRC client. Just point it to Libera.Chat with their link, [irc.libera.chat:6697](ircs://irc.libera.chat:6697), and you will find all the familiar faces in `#newpipe`. You can find more connection help on their [help site](https://libera.chat/guides/connect).

If you use Matrix, you will need to wait a bit until the Matrix team finalizes their bridge. NewPipe will continue support for Matrix though, and we will only fully close the freenode channel once Matrix is up and running.

## But I expected a new NewPipe version :(

> No worries, we will release a new version soon. And then link to it right here. It will be a calm release with a nice new feature, just the right thing after reading through this drama.

Reviewer: ```suggestion
No worries, we will release a new version soon. And then link to it right here. It will be a calm release with a nice new feature, just the thing we all need after reading through this drama.
```
